internal/parser/scanner/scanner.go

```diff
@@ -1,300 +1,21 @@
 package scanner
 
 import (
-	"fmt"
-
-	"github.com/tomarrell/lbadd/internal/parser/scanner/matcher"
 	"github.com/tomarrell/lbadd/internal/parser/scanner/token"
 )
 
+// sentinel allows constant errors
+type sentinel string
```

Comment:
Can we name this type something like `scannerError`? Sentinel is a bit vague.

Reply:
No, because this type of error is called sentinel error. A sentinel error is basically a package-global error that can be checked against. Problem is, most of them are variable (https://golang.org/src/io/io.go#L38). This allows someone to call `io.EOF = nil` and the whole world breaks down. `sentinel` is a type alias that allows constant errors. If we need it in more places, we should extract it to a separate package, allowing for errors like `const ErrMyError = sentinel.S("my error")`.
But no, we should not rename it.